src/shared/seccomp-util.c

```diff
@@ -790,37 +792,15 @@ int seccomp_restrict_namespaces(unsigned long retain) {
         if ((retain & NAMESPACE_FLAGS_ALL) == NAMESPACE_FLAGS_ALL)
                 return 0;
 
+#if CLONE_REVERSED_ORDER < 0
+#  warning "Consider adding the right clone() syscall definitions here!"
+#else
         SECCOMP_FOREACH_LOCAL_ARCH(arch) {
                 _cleanup_(seccomp_releasep) scmp_filter_ctx seccomp = NULL;
-                int clone_reversed_order = -1;
                 unsigned i;
 
                 log_debug("Operating on architecture: %s", seccomp_arch_to_string(arch));
 
-                switch (arch) {
-
-                case SCMP_ARCH_X86_64:
-                case SCMP_ARCH_X86:
-                case SCMP_ARCH_X32:
-                        clone_reversed_order = 0;
-                        break;
-
-                case SCMP_ARCH_S390:
-                case SCMP_ARCH_S390X:
-                        /* On s390/s390x the first two parameters to clone are switched */
-                        clone_reversed_order = 1;
-                        break;
-
-                /* Please add more definitions here, if you port systemd to other architectures! */
-
-#if !defined(__i386__) && !defined(__x86_64__) && !defined(__s390__) && !defined(__s390x__)
-#warning "Consider adding the right clone() syscall definitions here!"
-#endif
-                }
-
-                if (clone_reversed_order < 0) /* we don't know the right order, let's ignore this arch... */
-                        continue;
-
                 r = seccomp_init_for_arch(&seccomp, arch, SCMP_ACT_ALLOW);
```

Comment:
so I actually thought about this for a while and on purpose implemented the logic that the `SECCOMP_RESTRICT_NAMESPACES_BROKEN` macro was a compile-time thing, while the seccomp rules where a runtime-thing. The rationale behind this, was this:

The seccomp rules apply to a number of ABIs that the local CPU supports in parallel, and I didn't want to enforce any particular rules about that. However, the SECCOMP_RESTRICT_NAMESPACES_BROKEN thing was something that only our test would look for, the big distinction being in this case, that in the case of the tests we actually *use* a specific ABI (i.e. the ABI that our source tree is being compiled for) instead of just installing a filter for one. 

Now, this distinction doesn't matter too much of course. The only case where it could matter is in case there are CPUs supporting multiple ABIs that use a different clone() order on each... Right now, to my knowledge there is none, though, admitedly.

A theoretical benefit of my logic would be that we could generate filters with it for testing purposes with arbitrary archs set (i.e. even completely foreign ones), and then just see what filters seccomp actually generates for them. That's purely theoretical though...